### PR TITLE
RTEMIS-175: About us page theming.

### PR DESCRIPTION
**Pull request by sandee-mahlawat-innoraft**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-175

---
### What the PR does
`(Explain fixes, changes and impacts)`
- Fixes/changes/Impacts ...

---
### What I have tested
`(Explain usecases you have tested locally)`
- ex: State Admin: Configuation for USIDE code.
- ex: App Admin: Enables multilingual configuration.

---
### Extras
`(Add Screenshots or videos)`

### CODE REVIEW

File: modules/rte_mis_gin/rte_mis_gin.module [hunk 1]
```diff
@@ -47,21 +47,40 @@ function rte_mis_gin_page_attachments(array &$attachments) {
...
+  // Get the current path.
+  $current_path = \Drupal::service('path.current')->getPath();
+  // Get the alias of the current path.
+  $alias = \Drupal::service('path_alias.manager')->getAliasByPath($current_path);
+  // Add a suggestion for the front page.
   if ($route_name == 'view.frontpage.page_1') {
     $suggestions[] = 'page__front';
   }
+  // Add a suggestion for the /about path.
+  elseif ($alias == '/about') {
```
Comment (kunal-sin) [inline modules/rte_mis_gin/rte_mis_gin.module:61]: This should be `/about-us`

File: modules/rte_mis_gin/rte_mis_gin.module [hunk 2]
```diff
@@ -145,6 +168,10 @@ function rte_mis_gin_theme($existing, $type, $theme, $path) {
       'base hook' => 'page',
       'template' => 'page--front',
     ],
+    'generic_pages' => [
```
Comment (kunal-sin) [inline modules/rte_mis_gin/rte_mis_gin.module:171]: The name should pe `page__generic_pages`

File: modules/rte_mis_gin/js/about-us.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            $('.hamburger').on('click', function() {
+                $(this).toggleClass('active');
+                $('.main-menu-wrapper .menu').toggleClass('collapsed-menu');
+            });
+            // about us content wrapper
+            var $tempContainer = $('<div class="about-us-content-notification"></div>');
+            $('.about-us-wrapper .notifications-wrapper').appendTo($tempContainer);
+            $('.about-us-wrapper .about-us-content .layout-container').appendTo($tempContainer);
+            $tempContainer.wrapAll('<div class="about-us-content-notification-wrapper"></div>');
+            $('.about-us-wrapper').prepend($tempContainer.parent());
```
Comment (kunal-sin) [inline modules/rte_mis_gin/js/about-us.js:19]: Why we are adding divs using JS here?

File: modules/rte_mis_gin/js/about-us.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+(function ($, Drupal, once) {
+    Drupal.behaviors.wrapBlockMenu = {
+        attach: function (context, settings) {
+            $(once('wrapAll', '.content-wrapper .menu--social-links, .content-wrapper .menu--get-in-touch', context)).wrapAll('<div class="menu-social-links-wrapper block"></div>');
+            $(once('wrapAll', '.content-wrapper .block-system', context)).wrapAll('<div class="layout-container"></div>');
+            var $footerItems = $(once('wrapAll', '.content-wrapper .footer__site-logo-wrapper, .content-wrapper .menu--quick-links, .content-wrapper .menu--support, .content-wrapper .menu-social-links-wrapper, .content-wrapper .block-site__menu-text-section-block'), context);
+            $footerItems.wrapAll("<div class='footer-wrapper'></div>");
+            $(once('wrapAll', '.region-subheader-wrapper .block-site-logo-section-block, .region-subheader-wrapper .menu--main-menu, .region-subheader-wrapper .menu--account', context)).wrapAll('<div class="main-menu-wrapper"></div>');
```
Comment (kunal-sin) [inline modules/rte_mis_gin/js/about-us.js:8]: Why we are doing this via JS?

File: modules/rte_mis_gin/rte_mis_gin.module [hunk 5]
```diff
@@ -84,7 +103,11 @@ function rte_mis_gin_preprocess_html(array &$variables) {
         $variables['attributes']['class'][] = 'application-edit--view-wrapper';
       }
       break;
+  }
 
+  // Add a class if the current path is /about.
+  if ($alias == '/about') {
```
Comment (kunal-sin) [inline modules/rte_mis_gin/rte_mis_gin.module:109]: Change this to `/about-us`

File: modules/rte_mis_gin/js/about-us.js (hunk 3 above)
Comment (sandee-mahlawat-innoraft) [reply modules/rte_mis_gin/js/about-us.js:19]: For creating a wrapper on some items from different regions.

File: modules/rte_mis_gin/rte_mis_gin.module [hunk 6]
```diff
@@ -145,6 +168,10 @@ function rte_mis_gin_theme($existing, $type, $theme, $path) {
       'base hook' => 'page',
       'template' => 'page--front',
     ],
+    'generic_pages' => [
+      'base hook' => 'page',
+      'template' => 'page--generic-pages',
```
Comment (kunal-sin) [inline modules/rte_mis_gin/rte_mis_gin.module:173]: Can we change the name to `page--static-page.`

File: modules/rte_mis_gin/js/about-us.js (hunk 4 above)
Comment (sandee-mahlawat-innoraft) [reply modules/rte_mis_gin/js/about-us.js:8]: For creating a wrapper on some items from different regions, becuase we can't create any new regions.

Comment (kunal-sin) [reply modules/rte_mis_gin/js/about-us.js:8]: I can see that we are repeating this code in all the JS file, Can't we create a global JS for these things?

File: modules/rte_mis_home/rte_mis_home.module [hunk 7]
```diff
@@ -76,6 +93,11 @@ function rte_mis_home_preprocess_page(array &$variables) {
       ];
       break;
   }
+
+  // Add a class if the current path is /about-us.
+  if ($alias == '/about-us') {
+    $variables['notification'] = TRUE;
+  }
```
Comment (kunal-sin) [inline modules/rte_mis_home/rte_mis_home.module:100]: Why we are adding this in preprocess page hook?

File: modules/rte_mis_gin/templates/node--page.html.twig [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    </footer>
+  {% endif %}
+
+  <div{{ content_attributes.addClass('node__content') }}>
+    {{ content }}
+  </div>
+
+{% if notification %}
+    <div class="notifications-wrapper">
+      <h2>Notifications</h2>
```
Comment (kunal-sin) [inline modules/rte_mis_gin/templates/node--page.html.twig:106]: This should be translatable.

File: modules/rte_mis_gin/templates/node--page.html.twig [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+  {% if display_submitted %}
+    <footer class="node__meta">
+      {{ author_picture }}
+      <div{{ author_attributes.addClass('node__submitted') }}>
+        {% trans %}Submitted by {{ author_name }} on {{ date }}{% endtrans %}
+        {{ metadata }}
+      </div>
+    </footer>
+  {% endif %}
```
Comment (kunal-sin) [inline modules/rte_mis_gin/templates/node--page.html.twig:98]: Do we really need this?

File: modules/rte_mis_gin/js/about-us.js (hunk 3 above)
Comment (sandee-mahlawat-innoraft) [reply modules/rte_mis_gin/js/about-us.js:19]: No

Comment (sandee-mahlawat-innoraft) [reply modules/rte_mis_gin/js/about-us.js]: Removed this code.

File: modules/rte_mis_gin/templates/node--page.html.twig (hunk 9 above)
Comment (kumareshinnoraft) [reply modules/rte_mis_gin/templates/node--page.html.twig]: Removed the if statment block.

File: modules/rte_mis_gin/templates/node--page.html.twig (hunk 8 above)
Comment (kumareshinnoraft) [reply modules/rte_mis_gin/templates/node--page.html.twig]: Updated with the trans tag in twig.

File: modules/rte_mis_home/rte_mis_home.module (hunk 7 above)
Comment (kumareshinnoraft) [reply modules/rte_mis_home/rte_mis_home.module]: Removed the if block as it is already mentioned in the preprocess node.

**Pull request merged**